File: fixtures/scale/kotlin/large/src/cryptobench/registry/BlockRegistry.kt
package cryptobench.registry

import cryptobench.core.CipherSuite
import cryptobench.ciphers.feistel.FeistelSuite
import cryptobench.ciphers.tea.TeaSuite
import cryptobench.ciphers.xtea.XteaSuite
import cryptobench.ciphers.speck.SpeckSuite
import cryptobench.ciphers.simon.SimonSuite
import cryptobench.ciphers.ecb.EcbModeSuite
import cryptobench.ciphers.cbc.CbcModeSuite
import cryptobench.ciphers.ctr.CtrModeSuite

object BlockRegistry {
    fun suites(): List<CipherSuite> = listOf(
        FeistelSuite(),
        TeaSuite(),
        XteaSuite(),
        SpeckSuite(),
        SimonSuite(),
        EcbModeSuite(),
        CbcModeSuite(),
        CtrModeSuite(),
    )
}

File: fixtures/scale/dart/large/lib/registry/block_registry.dart
import '../core/cipher_suite.dart';
import '../ciphers/feistel/feistel_suite.dart';
import '../ciphers/tea/tea_suite.dart';
import '../ciphers/xtea/xtea_suite.dart';
import '../ciphers/speck/speck_suite.dart';
import '../ciphers/simon/simon_suite.dart';
import '../ciphers/ecb/ecb_suite.dart';
import '../ciphers/cbc/cbc_suite.dart';
import '../ciphers/ctr/ctr_suite.dart';

List<CipherSuite> blockSuites() => [
  FeistelSuite(),
  TeaSuite(),
  XteaSuite(),
  SpeckSuite(),
  SimonSuite(),
  EcbModeSuite(),
  CbcModeSuite(),
  CtrModeSuite(),
];
